src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java

```diff
@@ -75,6 +91,131 @@ public void adaptViewerTransform( Bookmark bookmark )
 		}
 	}
 
+	public void loadAdditionalBookmarks() {
+			Map<String, Bookmark> additionalBookmarks = bookmarksJsonParser.selectAndLoadBookmarks();
+			if (additionalBookmarks != null) {
+				nameToBookmark.putAll(additionalBookmarks);
+				bookmarkDropDown.removeAllItems();
+				for (String bookmarkName : nameToBookmark.keySet()) {
+					bookmarkDropDown.addItem(bookmarkName);
+				}
+			}
+	}
+
+	public void saveCurrentSettingsAsBookmark () {
+		ArrayList<String> bookmarkNameAndLocation = bookmarkSaveDialog();
+		Bookmark currentBookmark = getBookmarkFromCurrentSettings(bookmarkNameAndLocation.get(0));
+		ArrayList<Bookmark> bookmarks = new ArrayList<>();
+		bookmarks.add(currentBookmark);
+
+		if (bookmarkNameAndLocation.get(1).equals(FileUtils.PROJECT) &&
+				bookmarksJsonParser.getDatasetLocation().contains( "raw.githubusercontent" )) {
+			bookmarksJsonParser.saveBookmarksToGithub(bookmarks);
+		} else {
+			try {
+				bookmarksJsonParser.saveBookmarksToFile(bookmarks, bookmarkNameAndLocation.get(1));
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	private ArrayList<String> bookmarkSaveDialog () {
+		String fileLocation = null;
+		String bookmarkName = null;
+		final GenericDialog gd = new GenericDialog( "Choose save location" );
+		gd.addStringField("Bookmark Name", "name");
+		gd.addChoice( "Save to", new String[]{FileUtils.PROJECT, FileUtils.FILE_SYSTEM }, FileUtils.PROJECT );
+		gd.showDialog();
+		if ( gd.wasCanceled() ) return null;
+		bookmarkName = gd.getNextString();
+		fileLocation = gd.getNextChoice();
+
+		ArrayList<String> bookmarkNameandLocation = new ArrayList<>();
```

Comment:
This is confusing, because a `List` could have many items.

Just make a little non public class within the `BookmarksManger`

```
class NameAndLocation {
  String name;
  String location;
}
```